android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -142,4 +236,449 @@ public void rejectEvent(String eventId) {
+            return;
+        }
+
+        Radar.reverseGeocode(new Radar.RadarGeocodeCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarAddress[] addresses) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
+                        map.putString("status", status.toString());
+                        if (addresses != null) {
+                            map.putArray("addresses", RNRadarUtils.arrayForJson(RadarAddress.toJson(addresses)));
+                        }
+                        promise.resolve(map);
+                    } catch (JSONException e) {
+                        promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
+                    }
+                } else {
+                    promise.reject(status.toString(), status.toString());
+                }
+            }
+        });
+    }
+
+    @ReactMethod
+    public void reverseGeocode(ReadableMap locationMap, final Promise promise) {
+        if (promise == null) {
+            return;
+        }
+
+        double latitude = locationMap.getDouble("latitude");
+        double longitude = locationMap.getDouble("longitude");
+        Location location = new Location("RNRadarModule");
+        location.setLatitude(latitude);
+        location.setLongitude(longitude);
+
+        Radar.reverseGeocode(location, new Radar.RadarGeocodeCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarAddress[] addresses) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
+                        map.putString("status", status.toString());
+                        if (addresses != null) {
+                            map.putArray("addresses", RNRadarUtils.arrayForJson(RadarAddress.toJson(addresses)));
+                        }
+                        promise.resolve(map);
+                    } catch (JSONException e) {
+                        promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
+                    }
+                } else {
+                    promise.reject(status.toString(), status.toString());
+                }
+            }
+        });
+    }
+
+    @ReactMethod
+    public void ipGeocode(final Promise promise) {
+        if (promise == null) {
+            return;
+        }
+
+        Radar.ipGeocode(new Radar.RadarIpGeocodeCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarAddress address) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
```

Comment:
seems like we repeat this pattern quite often, not necessary for this version, but we can probably abstract the deserialization of a JSON key / checking nullability - ended up coming out pretty clean in the Python SDK https://github.com/radarlabs/radar-python/blob/5b9a61bda3e405565eb16d76b120cc27f7a2a7b3/radar/endpoints.py#L80-L84

Reply:
Yeah, we can refactor this in a second pass if needed